walletunlocker/service.go

```diff
@@ -100,18 +109,22 @@ type UnlockerService struct {
 	noFreelistSync bool
 	netParams      *chaincfg.Params
 	macaroonFiles  []string
+
+	quitChan <-chan struct{}
+	sync.Mutex
```

Comment:
nit: add a comment on this lock to indicate what it's for since we're not using it for every call (`GenSeed` doesn't acquire the lock).